porting.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
```

Comment:
#pragma once is not standard C++ if I'm correct...
Also, from what I understand, its semantics are not unambiguous.
Apart from that, the rest of the project uses #ifdef/#define/#endif. So please use that too, for consistency.